DigitalEcosystem/exploratory/exfoliation_bandgap_binning/exfoliation_bandgap_binning.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[]:
+
+
+import copy
+
+import pandas as pd
+import numpy as np
+import optuna
+import xgboost
+import imblearn.over_sampling
+import sklearn.model_selection
+import dscribe.descriptors
+import tqdm
+import sklearn.pipeline
+
+import functools
+
+import matplotlib.pyplot as plt
+import sklearn.impute
+import seaborn as sns
+
+import sys
+sys.path.append("../../../")
+import DigitalEcosystem.utils.figures
+
+tqdm.tqdm.pandas()
+
+
+# In[]:
+
+
+# Random seeds for reproducibility
+RANDOM_SEED = 1234
+import random
+random.seed(RANDOM_SEED)
+np.random.seed(RANDOM_SEED)
+
+
+# In[]:
+
+
+# Load up the data
+data_path = "../../refined/httpot/full_featurized_data.pkl"
+data = pd.read_pickle(data_path)
+
+cols_to_drop = ['formula',
+                'discovery_process (unitless)',
+                'potcars (unitless)',
+                'is_hubbard (unitless)',
+                'energy_per_atom (eV)',
+                'exfoliation_energy_per_atom (eV/atom)',
+                'is_bandgap_direct (unitless)',
+                'is_metal (unitless)',
+                'energy_vdw_per_atom (eV/atom)',
+                'total_magnetization (Bohr Magneton)']
+target_column = ['exfoliation_energy_per_atom (eV/atom)']
+matpedia_id = ['2dm_id (unitless)']
+atoms_col = ['atoms_object (unitless)']
+
+
+# In[]:
+
+
+noble_gases = ['He', 'Ne', 'Ar', 'Kr', 'Xe', 'Rn']
+fblock = ['La', 'Ce', 'Pr', 'Nd', 'Pm', 'Sm', 'Eu', 'Gd', 'Tb', 'Dy', 'Ho', 'Er', 'Tm', 'Yb', 'Lu',
+          'Ac', 'Th', 'Pa', 'U',  'Np', 'Pu', 'Am', 'Cm', 'Bk', 'Cf', 'Es', 'Fm', 'Md', 'No', 'Lr']
+d_synths = ['Rf', 'Db', 'Sg', 'Bh', 'HS', 'Mt', 'Ds', 'Rg', 'Cn', 'Nh', 'Fl', 'Mc', 'Lv', 'Ts', 'Og']
```

Comment:
These definitions should go into some central place and be reused across notebooks

Reply:
Moved